dev-tools/packaging/files/darwin/elastic-otel.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+BASEDIR=$(dirname "$0")
+
+exec $BASEDIR/elastic-agent otel $@
```

Comment:
For Darwin and Linux I believe we actually want
```
exec "$BASEDIR/elastic-agent" otel "$@"
```

If you put elastic-agent into a folder with a space and then cd to root and try to run it, it will fail due to the space without the path being quoted.

In addition I believe $@ needs to be in quotes:
>($@) Expands to the positional parameters, starting from one. When the expansion occurs within double quotes, each parameter expands to a separate word.

```
./a.sh 2 "3     4" 5
2 3     4 5                  # output for "$@"
2 3 4 5                      # output for $@  -> spaces are lost!
```

Reply:
Yes, run [shellcheck](https://www.shellcheck.net/) on this and it should find this problem and any other Bash quirks.